runtime/weak.c

```diff
@@ -221,7 +256,15 @@ CAMLexport void caml_ephemeron_unset_key(value ar, mlsize_t offset)
 
   offset += CAML_EPHE_FIRST_KEY;
 
-  do_check_key_clean(ar, offset);
+  if( caml_gc_phase == Phase_mark
+      && ephe_list_pure
+      && Field(ar, CAML_EPHE_DATA_OFFSET) != caml_ephe_none
```

Comment:
You could use `Is_white_during_mark` here, because if it's not white then `caml_darken` will do nothing.

Reply:
I meant using `Is_white_during_mark` instead of the `Field(...) != caml_ephe_none` test. But it doesn't seem important anyway.